src/context/AnimationsContext.js

```diff
@@ -0,0 +1,30 @@
+import PropTypes from 'prop-types';
+import React, { createContext, useContext, useState } from 'react';
+
+const AnimationsContext = createContext();
+
+export const useAnimation = () => useContext(AnimationsContext);
+const isBrowser = typeof window !== 'undefined';
+
+export const AnimationsProvider = ({ children }) => {
+  const [isPaused, setPaused] = useState(
+    isBrowser && localStorage.getItem('isPaused') === 'true'
+  );
+
+  const togglePaused = () => {
```

Comment:
`localStorage.setItem('isPaused', !isPaused);` powinno być wewnątrz setPaused ponieważ umieszczając to pod zmianą stanu, może się zdażyć tak, że do localstorage zostanie wrzucona błędna wartość. Aby temu zapowiedź zrobić to tak: 

```
const togglePaused = () => {
setPaused((prev) => {
if(isBrowser) localStorage.setItem('isPaused', !prev);
return !prev;
});
}
```

Reply:
Nie rozumiem ale zmienione! XD